tests/UnitTests/Services/UserServiceTest.cs

```diff
@@ -0,0 +1,219 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using Moq;
+using SnowBuddies.Application.AutoMapperConfiguration;
+using SnowBuddies.Application.Dtos;
+using SnowBuddies.Application.Implementation.Services;
+using SnowBuddies.Application.Interfaces.IRepositories;
+using SnowBuddies.Application.Interfaces.IServices;
+using SnowBuddies.Domain.Entities;
+using SnowBuddies.Infrastructure.Data;
+
+namespace UnitTests.Services
+{
+    public class UserServiceTest
+    {
+        private readonly IMapper _mapper;
+        private readonly Mock<IUserRepository> _mockUserRepository;
+        private IUserService _userService;
+
+        public UserServiceTest()
+        {
+            var profile = new ApplicationProfile();
+            var config = new MapperConfiguration(cfg => cfg.AddProfile(profile));
+            _mapper = new Mapper(config);
+            _mockUserRepository = new Mock<IUserRepository>();
+            _userService = new UserService(_mockUserRepository.Object, _mapper);
+        }
+
+        [Fact]
+        public async Task GetAllUsersAsync_ShouldReturnMappedUserDtos()
+        {
+            var userId = Guid.NewGuid();
+            var users = new List<User>()
```

Comment:
While I know Automapper tracks the relationship between User and UserDto objects, it seems like there's some interchange between the two in the tests, for example GetAllUsersAsync returns an IEnumerable of UserDtos. I'd like to keep consistency as much as possible for clarity.